.github/workflows/ca-publishing-crl-file-test.yml

```diff
@@ -166,82 +282,350 @@ jobs:
           docker exec pki openssl crl \
-              -in $FILENAME \
+              -in /var/lib/pki/pki-tomcat/crl/MasterCRL.crl \
               -inform DER \
               -text \
               -noout | tee output
 
-          # there should be one cert in the latest CRL
+          # CRL should contain two certs
           sed -n "s/^\s*\(Serial Number:.*\)\s*$/\1/p" output | wc -l > actual
-          echo "1" > expected
+          echo "2" > expected
+          diff expected actual
+
+      - name: Check user cert after revocation
+        run: |
+          # check user cert using OpenSSL
+          docker exec pki openssl verify \
+              -crl_check \
+              -crl_download \
+              -CAfile ca_signing.crt \
+              testuser.crt \
+              > >(tee stdout) 2> >(tee stderr >&2) || true
+
+          # user cert should be invalid
+          echo "UID = testuser" > expected
+          echo "error 23 at 0 depth lookup: certificate revoked" >> expected
+          echo "error testuser.crt: verification failed" >> expected
+          diff expected stderr
+
+          # check user cert using NSS with PKIX
+          docker exec pki /usr/lib64/nss/unsupported-tools/vfychain \
+              -d /root/.dogtag/nssdb \
+              -a \
+              -u 0 \
+              -pp \
+              -g leaf \
+              -m crl \
+              testuser.crt \
+              > >(tee stdout) 2> >(tee stderr >&2) || true
+
+          # user cert should be invalid
+          echo "Chain is bad!" > expected
+          head -1 stderr > actual
           diff expected actual
 
-      - name: Check CRL after cert unrevocation
+      - name: Check server cert after revocation
         run: |
-          docker exec pki /usr/share/pki/tests/ca/bin/ca-agent-cert-unrevoke.sh
+          # download the latest CRL
+          docker exec pki curl -sJO http://pki.example.com:8080/crl/MasterCRL.crl
+
+          # convert CRL to PEM
+          docker exec pki openssl crl \
+              -in MasterCRL.crl \
+              -inform DER \
+              -out MasterCRL.pem \
+              -outform PEM
+
+          # check server cert using OpenSSL
+          docker exec pki openssl verify \
+              -crl_check \
+              -CRLfile MasterCRL.pem \
+              -CAfile ca_signing.crt \
+              test.example.com.crt \
+              > >(tee stdout) 2> >(tee stderr >&2) || true
+
+          # server cert should be invalid
+          echo "CN = test.example.com" > expected
+          echo "error 23 at 0 depth lookup: certificate revoked" >> expected
+          echo "error test.example.com.crt: verification failed" >> expected
+          diff expected stderr
+
+          # import CRL into NSS
+          docker exec pki crlutil -I -d /root/.dogtag/nssdb -i MasterCRL.crl
+          docker exec pki crlutil -L -d /root/.dogtag/nssdb -n ca_signing
+
+          # check server cert using NSS without PKIX
```

Comment:
As above. Not sure if in this case, since CRL-DP is not present, we could test both PKIX and legacy and obtain the same result.